test/unit/test_linker.py

```diff
@@ -37,6 +39,25 @@ def test_linker_add_node(self):
 
         self.assertEqual(len(actual_nodes), len(expected_nodes))
 
+    def test_linker_write_and_read_graph(self):
+        expected_nodes = ['A', 'B', 'C']
+        for node in expected_nodes:
+            self.linker.add_node(node)
+
+        manifest = _mock_manifest('ABC')
+        (fd, fname) = tempfile.mkstemp()
+        self.linker.write_graph(fname, manifest)
+
+        new_linker = linker.from_file(fname)
+        os.close(fd)
+        os.unlink(fname)
```

Comment:
Can you wrap this unlink in a `finally` block? Something like this, probably:
```python
        (fd, fname) = tempfile.mkstemp()
        os.close(fd)
        try:
            self.linker.write_graph(fname, manifest)
            new_linker = linker.from_file(fname)
        finally:
            os.unlink(fname)
```

As it is, on failure this will leave files in /tmp

Reply:
Thanks for the suggestion! I've added this change to the feature branch.

(Sorry for not getting back sooner -- I was on vacation until today)